Doc-Tests are failing

On the current `main` (v.1.1.4) running

```
$ cargo test --workspace
```

shows some failures in Doc-Tests:

```
   Doc-tests huff_neo_codegen

running 4 tests
test crates/codegen/src/lib.rs - (line 26) ... FAILED
test crates/codegen/src/lib.rs - (line 109) ... FAILED
test crates/codegen/src/lib.rs - (line 53) ... FAILED
test crates/codegen/src/lib.rs - Codegen (line 39) ... ok

failures:

---- crates/codegen/src/lib.rs - (line 26) stdout ----
error[E0432]: unresolved import `huff_neo_utils::files`
 --> crates/codegen/src/lib.rs:29:21
  |
6 | use huff_neo_utils::files::FileSource;
  |                     ^^^^^ could not find `files` in `huff_neo_utils`

error: aborting due to 1 previous error

For more information about this error, try `rustc --explain E0432`.
Couldn't compile the test.
---- crates/codegen/src/lib.rs - (line 109) stdout ----
error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:125:17
   |
19 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:129:17
   |
23 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:133:17
   |
27 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:137:17
   |
31 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:142:13
   |
36 |       span: AstSpan(vec![]),
   |             ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0433]: failed to resolve: use of undeclared type `HashSet`
  --> crates/codegen/src/lib.rs:154:11
   |
48 |   labels: HashSet::new(),
   |           ^^^^^^^ use of undeclared type `HashSet`
   |
help: consider importing one of these items
   |
2  + use std::collections::HashSet;
   |
2  + use alloy_primitives::map::HashSet;
   |
2  + use alloy_primitives::map::hash_set::HashSet;
   |

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
Couldn't compile the test.
---- crates/codegen/src/lib.rs - (line 53) stdout ----
error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:69:17
   |
19 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:73:17
   |
23 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:77:17
   |
27 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:81:17
   |
31 |           span: AstSpan(vec![]),
   |                 ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0425]: cannot find function, tuple struct or tuple variant `AstSpan` in this scope
  --> crates/codegen/src/lib.rs:86:13
   |
36 |       span: AstSpan(vec![]),
   |             ^^^^^^^ not found in this scope
   |
help: consider importing this tuple struct
   |
2  + use huff_neo_utils::ast::span::AstSpan;
   |

error[E0433]: failed to resolve: use of undeclared type `HashSet`
  --> crates/codegen/src/lib.rs:98:11
   |
48 |   labels: HashSet::new(),
   |           ^^^^^^^ use of undeclared type `HashSet`
   |
help: consider importing one of these items
   |
2  + use std::collections::HashSet;
   |
2  + use alloy_primitives::map::HashSet;
   |
2  + use alloy_primitives::map::hash_set::HashSet;
   |

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0425, E0433.
For more information about an error, try `rustc --explain E0425`.
Couldn't compile the test.

failures:
    crates/codegen/src/lib.rs - (line 109)
    crates/codegen/src/lib.rs - (line 26)
    crates/codegen/src/lib.rs - (line 53)

test result: FAILED. 1 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.42s

error: doctest failed, to rerun pass `-p huff-neo-codegen --doc`
```

--- a/crates/codegen/README.md
+++ b/crates/codegen/README.md
@@ -25,7 +25,7 @@ Below we showcase generating a compile artifact from compiled bytecode using `hu
 
 ```rust
 use huff_neo_codegen::*;
-use huff_neo_utils::files::FileSource;
+use huff_neo_utils::prelude::*;
 use std::sync::Arc;
 use std::cell::RefCell;
 use std::rc::Rc;
@@ -53,6 +53,7 @@ Let's say you have a [Contract](../utils/ast/struct.Contract.html) instance with
 ```rust
 use huff_neo_codegen::*;
 use huff_neo_utils::prelude::*;
+use std::collections::HashSet;
 use std::sync::{Arc, Mutex};
 
 // Mock contract with a main macro
@@ -109,6 +110,7 @@ Similarly, once you have a [Contract](../utils/ast/struct.Contract.html) instanc
 ```rust
 use huff_neo_codegen::*;
 use huff_neo_utils::prelude::*;
+use std::collections::HashSet;
 use std::sync::{Arc, Mutex};
 
 // Mock contract with a constructor macro

--- a/crates/lexer/README.md
+++ b/crates/lexer/README.md
@@ -24,7 +24,7 @@ use std::ops::Deref;
 // Instantiate a new lexer
 let source = "#define macro HELLO_WORLD()";
 let flattened_source = FullFileSource { source, file: None, spans: vec![] };
-let mut lexer = Lexer::new(flattened_source.source);
+let mut lexer = Lexer::new(flattened_source);
 
 // This token should be a Define identifier
 let tok = lexer.next().unwrap().unwrap();

--- a/crates/parser/README.md
+++ b/crates/parser/README.md
@@ -19,12 +19,13 @@ definition.
 use huff_neo_utils::prelude::*;
 use huff_neo_lexer::{Lexer};
 use huff_neo_parser::{Parser};
+use std::collections::HashSet;
 use std::sync::{Arc, Mutex};
 
 // Create a Lexer from the source code
 let source = "#define macro HELLO_WORLD() = takes(0) returns(0) {}";
 let flattened_source = FullFileSource { source, file: None, spans: vec![] };
-let mut lexer = Lexer::new(flattened_source.source);
+let mut lexer = Lexer::new(flattened_source);
 
 // Grab the tokens from the lexer
 let tokens = lexer.into_iter().map(|x| x.unwrap()).collect::<Vec<Token>>();

--- a/crates/utils/src/lib.rs
+++ b/crates/utils/src/lib.rs
@@ -56,9 +56,10 @@ pub mod shell;
 
 /// Prelude wraps common utilities.
 pub mod prelude {
-    pub use crate::ast::huff::*;
+    pub use crate::ast::{abi::*, huff::*, span::*};
     pub use crate::file::span::*;
     pub use crate::file::unpack_files::*;
+    pub use crate::file::{file_source::*, full_file_source::*};
     pub use crate::{
         abi::*, artifact::*, bytecode::*, bytes_util::*, error::*, evm_version::*, opcodes::*, report::*, sol_interface::*, token::*,
         types::*,

